examples/test_data_gen/src/construct_test_data_flow/utils.py

```diff
@@ -0,0 +1,123 @@
+import json
+import re
+from collections import namedtuple
+from enum import Enum
+
+import numpy as np
+import numpy.testing as npt
+from numpy.random import default_rng
+
+from promptflow.connections import AzureOpenAIConnection, OpenAIConnection
+from promptflow.tools.aoai import chat as aoai_chat
+from promptflow.tools.openai import chat as openai_chat
+
+
+class QuestionType:
+    SIMPLE = "simple"
+    REASONING = "reasoning"
+    CONDITIONAL = "conditional"
+    # MULTI_CONTEXT = "multi_context"
+
+
+class ValidateObj:
+    QUESTION = "question"
+    TEXT_TRUNK = "text_trunk"
+    GROUND_TRUTH = "ground_truth"
+
+
+class ResponseFormat(Enum):
+    TEXT = "text"
+    JSON = "json_object"
+
+
+class ErrorMsg:
+    INVALID_JSON_FORMAT = "llm failed to return the verdict and reason in correct json format. Response: {0}"
+    INVALID_TEXT_TRUNK = "Skipping generating seed question due to invalid text chunk."
+    INVALID_QUESTION = "Invalid seed question: {0}"
+    INVALID_ANSWER = "Invalid answer: {0}"
+
+
+ValidationResult = namedtuple("ValidationResult", ["pass_validation", "reason_if_failed"])
+
+
+def llm_call(connection, model, prompt, response_format="text"):
+    if isinstance(connection, AzureOpenAIConnection):
```

Comment:
please avoid unnecessary jinja render in python tool, which may cause jinja rendering failure.
```
prompt = f"{{% raw %}}{prompt}{{% endraw %}}"
```